lib/gitlib.sh

```diff
@@ -262,3 +262,75 @@ gitlib::repo_state () {
     return 1
   fi
 }
+
+# -----------------------------------------------------------------------------
```

Comment:
This is copied from kubernetes/hack/lib/version.sh.  Why can't we use push-build.sh in the same way the CI does, agnostically, calling it from the scope of the build directory?

Reply:
`kubernetes/hack/lib/version.sh` is from a different repo and if i source that script in push-build.sh which is in release repo. Then its like creating a dependency on kubernetes repo for release repo. we need the capability of getting the version of the build, but we should avoid depending on kubernetes repo/release.

Currently the version of the build is fetched from `cluster/kubectl.sh version --client` which is not generic. In federation repo/release we do not have kubectl. So in order to be generic i use `git` directly through the `get_version_vars` function copied from `kubernetes/hack/lib/version.sh`